### ISSUE TITLE
Freezes on startup at "Cross referencing miner device IDs...."; killing nbminer process resumes startup

### ISSUE THREAD

**Issue by damonlynch** [nicehash/NiceHashMiner#1702]
On one of my machines (AMD X470 / Ryzen 3 / 64 GB RAM / Windows 10 1903 with all updates), Nicehash 1.9.2.10 freezes on startup. The status message directly above the NH window  says `Cross referencing miner device IDs....` Killing the nbminer process using task manager resumes startup. Before killing the process, the latest log entry was:

`[2019-07-29 23:19:41] [INFO] [SOCKET] Received: {"method":"balance","value":"0.000039"}.` 

Attached is the log file (with NH currently running normally in the background)
[log.txt](https://github.com/nicehash/NiceHashMiner/files/3445056/log.txt)

Thank you.

**Comment by damonlynch** [comment]
Can confirm this bug still occurs with version 1.9.2.11.

**Comment by Bob-knarley** [comment]
I've got the same issue (INTEL 7700 / 1080 ti / Windows 10  (should be latest version))

**Comment by luc1an24** [comment]
Hi @damonlynch 
Thank you for reaching out.
Could you please provide the logs from when the bug occurs?
So delete/rename previous logs -> start NHM -> wait for freeze -> close NHM -> send logs.
Thank you very much!

**Comment by damonlynch** [comment]
Hi @luc1an24 
Attached is the logfile. At this point of the logfile, the Nicehash has just closed after I clicked the X on the right of the window's titlebar, as you instructed.  nbminer.exe is still running in the background.
[log.txt](https://github.com/nicehash/NiceHashMiner/files/3481891/log.txt)

**Comment by luc1an24** [comment]
Thanks for logs.
If you wait for a minute does nbminer closes by itself or it hangs.
Also if you close the nbminer from task manager are you then able to mine?
Logs are not showing any problems so this is probably the issue with miner. Did you experience any other miner causing the freeze during the `Cross referencing` step?
Thank you

**Comment by damonlynch** [comment]
If you wait for an hour nbminer is still running and NiceHash is still displaying the message "Cross referencing miner device IDs...". 

Only when manually closing nbminer with the task manager does mining begin.

No other miner has exhibited this problem.

**Comment by Bob-knarley** [comment]
I've got the exact same thing

**Comment by luc1an24** [comment]
There seems to be a problem with the compatibility of miner and system.
Could you please run `CreateLogReport.exe` and send me the archived logs?
You can send them to `domen.kirnkrefl@nicehash.com` so no personal data will be public.
Thank you for your help.

**Comment by damonlynch** [comment]
Log report has been sent by email

**Comment by luc1an24** [comment]
Could you please run following build (improved logging) so I can get more data about where exactly freeze happens: 

[NiceHashMiner_log.zip](https://github.com/nicehash/NiceHashMiner/files/3485486/NiceHashMiner_log.zip)


After freeze please send log.txt to me

**Comment by damonlynch** [comment]
The first time I ran the updated miner, it actually started up normally (the first time NHM has done that for a few weeks, from memory). However the second time I started it, the freeze came back. I have attached two versions of the log file: one at  the point of the freeze (`log_before_nbminer_process_end.txt`), and the second after ending the the nbminer process (`log.txt`).

[log_before_nbminer_process_end.txt](https://github.com/nicehash/NiceHashMiner/files/3487097/log_before_nbminer_process_end.txt)
[log.txt](https://github.com/nicehash/NiceHashMiner/files/3487099/log.txt)

**Comment by luc1an24** [comment]
In these logs I can't find any trace of mining with NBMiner.
Did you attach right logs?
So Cross referencing now goes through but it freezes during mining?

**Comment by damonlynch** [comment]
I did attach the correct logs. The cross referencing does not finish until I manually end the NBminer process. Screenshot is attached, before I end the NBminer process. Please note at this point at that moment NHM had been frozen for about a minute or so while I figured out how to use the built-in windows 10 screenshot tool.

![NBMiner-Nicehash-freeze](https://user-images.githubusercontent.com/1749868/62866759-a98f9900-bcdf-11e9-9537-ed04ac25bc53.png)

**Comment by binary8** [comment]
Hello, I hope this finds you well. did you get an answer to this? I have NiceHash install doing the same issue and have tried all I feel like - I added miner.exe to safe files in defender and my AVR as well as turned everything on and off, restarted the install, how did you get your Nice Hash to complete install? Mine worked first time then stopped when I tried to reinstall after computer reset. TIA!!!!

**Comment by abdmous** [comment]
Hello, were you able to find a solution for this problem ? I have the same Issue :)
Thank you

**Comment by Bob-knarley** [comment]
I believe they updated the miner and it fixed the issue for me, either that
or I had to add another exception for it in my antivirus.

**Comment by damonlynch** [comment]
@abdmous I uninstall NBminer on my desktop computer whenever I install NiceHash. I do that not because of the bug in this bug report, but because I face a different problem with NBMiner:  whenever I play any video content (e.g. YouTube, VLC player), NBMiner performance becomes either terrible or simply stops working altogether. I have two other systems on which I use NiceHash, and they run NBMiner without difficulties.

**Comment by jemtsawtinhwa** [comment]
Same issue happen here.

**Comment by usvmining-2021** [comment]
Hi, same issue here crash at 93%
My rig setup:
Windows 10 x64
4GB RAM
8x RX570 4GB
GPU Driver: Adrenalin 2019 19.75
MSI AB settings : vcore: -100, mclock: 1900, cclock: 1000, temp: 80

[log-02012021.txt](https://github.com/nicehash/NiceHashMiner/files/5759804/log-02012021.txt)

**Comment by xberg** [comment]
Exactly the same problem here March 2021. 
1st run of Nicehash Miner. Blocked at 92% "Cross referencing miner device IDs" / and 66% cross reference NBMiner
I do not see NBMiner running in the task manager: I do not know of a workaround to get NH to work.
Nicehash QuickMiner works OK.

**Comment by yegorvin** [comment]
Same problem on startup.
"cross referencing miner device ids" -> "Cross Reference NBMiner"

![nice](https://user-images.githubusercontent.com/59833019/145677135-37e58e71-09e8-4f7d-afad-1d4fb4d29dea.JPG)
[log.zip](https://github.com/nicehash/NiceHashMiner/files/7697044/log.zip)

If delete nbminer manually - NiceHash started ok.

Reinstall program does not help.

**Comment by RafaelProgramming** [comment]
same error happens to me, the screen stops at 50%, usando version NiceHash Miner v3.0.7.2

![Travou](https://user-images.githubusercontent.com/100859464/160663181-ecbad26f-ac83-49d7-9bac-56216165068a.png)

I found the problem, removed the plugin, nbminer, and started without any problems with all other plugins, when I installed nbminer again, the screen stopped at 50%, but even with the screen locked, if left alone, nbminer, starts, only, nothing appears on the screen, it is locked. ah some error in the program because with all other plugins, they work. I just wait for this nbminer plugin to be corrected and mine again correctly, appearing on the screen like all the other plugins.
Now, if you enter the folder, minerplugins, and remove the folder from nbminer, close the program and open it again, the program starts normally

[log.txt](https://github.com/nicehash/NiceHashMiner/files/8376306/log.txt)
